lisa/sut_orchestrator/azure/tools.py

```diff
@@ -579,3 +580,91 @@ def get_pool_records(self, pool_id: int, force_run: bool = False) -> Dict[str, s
             records[content_split[i]] = content_split[i + 1]
 
         return records
+
+
+class BlobDownloader(Tool):
+    @property
+    def command(self) -> str:
+        return "echo blob_downloader"
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def _check_exists(self) -> bool:
+        return True
+
+    def download_blob(
+        self,
+        account_name: str,
+        container_name: str,
+        blob_name: str,
+        credential: Optional[Any] = None,
+        subscription_id: Optional[str] = None,
+        cloud: Optional[Cloud] = None,
+        resource_group_name: Optional[str] = None,
+        connection_string: Optional[str] = None,
+    ) -> PurePath:
+        from lisa import constants
+        from lisa.sut_orchestrator.azure.common import get_or_create_storage_container
+
+        working_path = constants.RUN_LOCAL_WORKING_PATH
+        working_path.mkdir(parents=True, exist_ok=True)
+        file_path = working_path / blob_name
+        container_client = get_or_create_storage_container(
+            container_name,
+            credential,
+            subscription_id,
+            cloud,
+            account_name,
+            resource_group_name,
+            connection_string,
+            False,
+        )
+        blob_client = container_client.get_blob_client(blob_name)
+        if not blob_client.exists():
+            raise LisaException(
+                f"Blob {blob_name} not found in container {container_name}"
+            )
+        try:
+            blob_data = blob_client.download_blob()
+            with open(file_path, "wb") as f:
+                f.write(blob_data.readall())
+            self._log.debug("Blob downloaded successfully.")
+        except Exception as e:
+            raise LisaException("An error occurred during blob download.") from e
+        return file_path
+
+    def list_blobs(
```

Comment:
Is it used somewhere? Change them use this method.

Reply:
I put download and list logic into azure common file, and refer from there